mlir/include/mlir/Dialect/Utils/StaticValueUtils.h

```diff
@@ -200,6 +200,12 @@ struct SaturatedInteger {
     return SaturatedInteger{false, other.v + v};
   }
   SaturatedInteger operator*(SaturatedInteger other) {
+    // Multiplication with 0 is always 0.
+    if (!other.saturated && other.v == 0)
+      return SaturatedInteger{false, 0};
```

Comment:
TIL `SaturatedInteger`; am I right in reading the definition that "saturation" connotes dynamic rather than concrete? If so, a little odd.

Reply:
Yes, that's right. Not sure why it's called "saturated".